cfme/infrastructure/provider/__init__.py

```diff
@@ -37,56 +29,23 @@
 from utils.wait import wait_for
 
 match_page = partial(match_location, controller='ems_infra', title='Infrastructure Providers')
-
-properties_form = TabStripForm(
-    fields=[
-        ('type_select', AngularSelect("emstype")),
-        ('name_text', Input("name")),
-        ("api_version", AngularSelect("api_version")),
-    ],
-    tab_fields={
-        "Default": [
-            ('hostname_text', Input("default_hostname")),
-            ('api_port', Input("default_api_port")),
-            ('sec_protocol', AngularSelect("default_security_protocol", exact=True)),
-            ('verify_tls_switch', BootstrapSwitch(input_id="default_tls_verify")),
-            ('ca_certs', Input('default_tls_ca_certs')),
-        ],
-        "Events": [
-            ('event_selection', Radio('event_stream_selection')),
-            ('amqp_hostname_text', Input("amqp_hostname")),
-            ('amqp_api_port', Input("amqp_api_port")),
-            ('amqp_sec_protocol', AngularSelect("amqp_security_protocol", exact=True)),
-        ],
-        "C & U Database": [
-            ('candu_hostname_text', Input("metrics_hostname")),
-            ('acandu_api_port', Input("metrics_api_port")),
-        ]
-    })
-
-prop_region = Region(locators={'properties_form': properties_form})
-
-cfg_btn = partial(tb.select, 'Configuration')
 pol_btn = partial(tb.select, 'Policy')
-mon_btn = partial(tb.select, 'Monitoring')
 
 
-class InfraProvider(Pretty, CloudInfraProvider):
+class InfraProvider(Pretty, CloudInfraProvider, Fillable):
     """
     Abstract model of an infrastructure provider in cfme. See VMwareProvider or RHEVMProvider.
 
     Args:
         name: Name of the provider.
         details: a details record (see VMwareDetails, RHEVMDetails inner class).
-        credentials (:py:class:`Credential`): see Credential inner class.
         key: The CFME key of the provider in the yaml.
-        candu: C&U credentials if this is a RHEVMDetails class.
 
     Usage:
 
         myprov = VMwareProvider(name='foo',
-                             region='us-west-1',
-                             credentials=Provider.Credential(principal='admin', secret='foobar'))
+                             region='us-west-1'
+                             endpoints=SomeEndpoint())
```

Comment:
**REQUIRED:** Don't we need to support multiple endpoints? Shouldn't this be a dict of some kind? OK after looking below I see what you do, we should mention in the comments above how this works.